src/Data/IntegerInterval.hs

```diff
@@ -309,6 +310,17 @@ width x
         (Finite lb, Finite ub) -> ub - lb
         _ -> error "Data.IntegerInterval.width: unbounded interval"
 
+-- | How many integers lie within the (bounded) interval.
+-- Equal to @Just (width + 1)@ for non-empty, bounded intervals.
+-- The @memberCount@ of an unbounded interval is @Nothing@.
+memberCount :: IntegerInterval -> Maybe Integer
```

Comment:
As usual, the shorter PR the longer bikeshedding :) 

I wonder if `Maybe Natural` would be a better return type. Two benefits:
* It's the smallest possible set of values, making `memberCount` surjective.
* It's more distinguishing, e. g., one cannot replace `memberCount` with `pickup`.
An obvious penalty:
* Users are likely to need more `fromIntegral` / `toInteger`.

I'm split. @ncfavier what do you think? Let's do what you prefer.

Reply:
If this were Agda I would definitely insist on returning `Maybe ℕ`, but in a low-level language like Haskell I don't think it's worth the trouble. You don't pattern match on `Integer`.